lib/ansible/plugins/vars/host_group_vars.py

```diff
@@ -128,7 +134,7 @@ def _get_dir_files(self, path):
 
                 if os.path.isdir(full_spath) and not ext:  # recursive search if dir
                     found.extend(self._get_dir_files(full_spath))
-                elif os.path.isfile(full_spath) and (not ext or ext in C.YAML_FILENAME_EXTENSIONS):
+                elif os.path.isfile(full_spath) and (not ext or ext in C.YAML_FILENAME_EXTENSIONS) and not self._is_ext_blacklisted(full_spath):
```

Comment:
this is redundant as if `(not ext or ext in C.YAML_FILENAME_EXTENSIONS)`  if True means it is not a blacklisted extension, if false .. no need to check blacklisted.